pyvista/core/utilities/geometric_objects.py

```diff
@@ -383,6 +395,521 @@ def Sphere(
+    >>> solid_sphere.plot(
+    ...     scalars="Volume", show_edges=True, clim=[3e-5, 5e-4]
+    ... )
+
+    Sampling the polar angle in a nonlinear manner allows for consistent cell volumes.  See
+    `Sphere Point Picking <https://mathworld.wolfram.com/SpherePointPicking.html>`_.
+
+    >>> phi = np.rad2deg(np.arccos(np.linspace(1, -1, 30)))
+    >>> solid_sphere = pv.SolidSphereGeneric(
+    ...     radius=np.linspace(0, 0.5, 2),
+    ...     theta=np.linspace(180, 360, 30),
+    ...     phi=phi,
+    ... )
+    >>> solid_sphere = solid_sphere.compute_cell_sizes()
+    >>> solid_sphere.plot(
+    ...     scalars="Volume", show_edges=True, clim=[3e-5, 5e-4]
+    ... )
+
+    """
+    if radius is None:
+        radius = np.linspace(0, 0.5, 5)
+    radius = np.asanyarray(radius)
+
+    # Hereafter all degrees are in radians
+    # Also radius, phi, theta are now np.ndarrays
+    if theta is None:
+        theta = np.linspace(0, 2 * np.pi, 30)
+    else:
+        if radians:
+            theta = np.asanyarray(theta)
+        else:
+            theta = np.deg2rad(theta)
+
+    if phi is None:
+        phi = np.linspace(0, np.pi, 30)
+    else:
+        if radians:
+            theta = np.asanyarray(theta)
+        else:
+            phi = np.deg2rad(phi)
+
+    nr = len(radius)
+    ntheta = len(theta)
+    nphi = len(phi)
+
+    if nr < 2:
+        raise ValueError("radius resolution must be 2 or more")
+    if ntheta < 2:
+        raise ValueError("theta resolution must be 2 or more")
+    if nphi < 2:
+        raise ValueError("phi resolution must be 2 or more")
+
+    def _is_sorted(a):
+        return np.all(a[:-1] < a[1:])
+
+    if not _is_sorted(radius):
+        raise ValueError("radius is not monotonically increasing")
+    if not _is_sorted(theta):
+        raise ValueError("theta is not monotonically increasing")
+    if not _is_sorted(phi):
+        raise ValueError("phi is not monotonically increasing")
+
+    def _greater_than_equal_or_close(value1, value2):
+        return value1 >= value2 or np.isclose(value1, value2, rtol=0.0, atol=tolerance)
+
+    def _less_than_equal_or_close(value1, value2):
+        return value1 <= value2 or np.isclose(value1, value2, rtol=0.0, atol=tolerance)
+
+    if not _greater_than_equal_or_close(radius[0], 0.0):
+        raise ValueError("minimum radius cannot be negative")
+
+    # range of theta cannot be greater than 360 degrees
+    if not _less_than_equal_or_close(theta[-1] - theta[0], 2 * np.pi):
+        if radians:
+            raise ValueError("max theta and min theta must be within 2 * np.pi")
+        raise ValueError("max theta and min theta must be within 360 degrees")
+
+    if not _greater_than_equal_or_close(phi[0], 0.0):
+        raise ValueError("minimum phi cannot be negative")
+    if not _less_than_equal_or_close(phi[-1], np.pi):
+        if radians:
+            raise ValueError("maximum phi cannot be > np.pi")
+        raise ValueError("maximum phi cannot be > 180.0")
```

Comment:
```suggestion
        raise ValueError("maximum phi cannot be > " + "np.pi" if radians else "180.0")
```

Reply:
Same as other similar comment.

Reply:
```suggestion
        if radians:
            max_angle = "np.pi"
        else:
            max_angle = "180"
        raise ValueError(f"maximum phi cannot be {max_angle}")
```

Reply:
@tkoyama010 I've edited your suggestion to include the `>`, heads-up :)

Reply:
```suggestion
        max_angle = "np.pi" if radians else "180 degrees"
        raise ValueError(f"maximum phi cannot be > {max_angle}")
```